blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md

```diff
@@ -0,0 +1,1195 @@
+
+Next, To interact with the cUSD ERC-20 token on the Celo alfajores test network, you need to know the address of the token.
+
+
+After defining the single variables used in the contract, you want to create a "model" for a car, give the car its properties and group variables together.
+
+```js
+    struct Car{
+        address payable owner;
+        string carName;
+        string carDescription;
+        string carImage; 
+        uint price;
+        bool isUsed; 
+        bool isBought;
+    }
+    
+    mapping (uint => Car) internal cars;
+```
+
+To do this, you would require a struct data type with the keyword `struct` and give it multiple properties. ([Learn about structs here](https://docs.soliditylang.org/en/latest/types.html#structs))
+
+For this tutorial, these would be the variables that you would store in the struct:
+1. owner - This would store the address of the owner of a particular car as all cars in the marketplace would be owned by someone. It is of the address type
+2. carName - This stores the name of the car. It is of type string
+3. carImage - stores the url of the car. Type string
+4. carDescription - car description.  Type string
+5. price - This stores the price of the car. Its a number so its of type uint
+6. isUsed - This would store if the car is used or not. It would be a boolean type. True or false
+7. isSale - This keeps track of when a car is on sale or not. Boolean type
+8. isBought - This keeps track of when a car has been bought. Boolean type
+
+
+To handle multiple cars, a mapping is needed where you can access the value of a car through their key. Just like dictionaries in python or objects in Javascript.
+To create a mapping, you use the keyword `mapping` and assign a key type to a value type. In this case, your key would be an integer and the value would be the struct Car we just created.
+
+
+In the next section, you will define a function to add the car to the smart contract.
+
+```js
+    function setCar(
+        string memory _carName,
+        string memory _carDescription,
+        string memory _carimage,
+        bool _isUsed,
+        uint _price
+    )public {
+        cars[carLength] = Car(
+              payable(msg.sender),
+              _carName,
+              _carDescription,
+              _carimage,
+              _price,
+              _isUsed,
+              false
+        );
+        carLength++;
+    }
+```
+
+You have to specify the parameters type in the function. In this case, we need to pass the name, description, image, price and isUsed (all with an underscore to differentiate them from the struct values) as parameters to the function. Name the function `setCar` with a visibility of public.
```

Comment:
THE TYPE OF PARAMETERS(the parameters type)-167 line 1

remove a (with visibility)-line 167- 3